src/frontend/js/widgets/Dashboard/components/DashboardItem/Order/Contract/index.tsx

```diff
@@ -0,0 +1,149 @@
+import { defineMessages, FormattedMessage, useIntl } from 'react-intl';
+import { Button, Loader } from '@openfun/cunningham-react';
+import { DashboardItem } from 'widgets/Dashboard/components/DashboardItem/index';
+import { Contract, CredentialOrder, Product } from 'types/Joanie';
+import { RouterButton } from 'widgets/Dashboard/components/RouterButton';
+import { LearnerDashboardPaths } from 'widgets/Dashboard/utils/learnerRouteMessages';
+import { getDashboardRoutePath } from 'widgets/Dashboard/utils/dashboardRoutes';
+import { useJoanieApi } from 'contexts/JoanieApiContext';
+import { alert } from 'utils/indirection/window';
+
+const messages = defineMessages({
+  contractSigned: {
+    id: 'components.DashboardItemOrder.contractSigned',
+    description:
+      "Message displayed when the order's contract has been signed and can be downloaded.",
+    defaultMessage: "You've accepted the training contract.",
+  },
+  contractSignActionLabel: {
+    id: 'components.DashboardItemOrder.contractSignActionLabel',
+    description: 'Label of "sign contract" action.',
+    defaultMessage: 'Sign',
+  },
+  contractDownloadActionLabel: {
+    id: 'components.DashboardItemOrder.contractDownloadActionLabel',
+    description: 'Label of "download contract" action.',
+    defaultMessage: 'Download',
+  },
+  contractUnsigned: {
+    id: 'components.DashboardItemOrder.contractUnsigned',
+    description: "Message displayed when the order's contract needs to be signed.",
+    defaultMessage: 'You have to sign this contract to access your training.',
+  },
+  contractDownloadError: {
+    id: 'components.DashboardItemOrder.contractDownloadError',
+    description: "Message displayed when the order's contract download fails.",
+    defaultMessage: 'An error happened while downloading the contract. Please try again later.',
+  },
+});
+
+export const DashboardItemOrderContractFooter = ({
+  order,
+  contract,
+  onSign,
+  loading,
+  writable,
+}: {
+  order: CredentialOrder;
+  contract?: Contract;
+  mode?: 'default' | 'compact';
+  onSign?: () => void;
+  loading?: boolean;
+  writable: boolean;
+}) => {
+  const getRoutePath = getDashboardRoutePath(useIntl());
+  const api = useJoanieApi();
+  const intl = useIntl();
+
+  const downloadContract = async () => {
+    try {
+      const blob = await api.user.contracts.download(contract!.id);
+      // eslint-disable-next-line compat/compat
+      const file = window.URL.createObjectURL(blob);
+      window.open(file);
+
+      // eslint-disable-next-line compat/compat
+      URL.revokeObjectURL(file);
+    } catch (e) {
+      alert(intl.formatMessage(messages.contractDownloadError));
```

Comment:
I think in this case, we should also dispatch an error to sentry.

```suggestion
      handle(e)
      alert(intl.formatMessage(messages.contractDownloadError));
```